### DISCUSSION TITLE
Terminating Condition if we don't have data in db table and generate Jfree Chart if it contains data

### DISCUSSION THREAD

**Discussion by nagamalli148** [jfree/jfreechart#303] [Q&A]
I need a condition which should terminate if there's no data and redirects to chart generation if data is available in jfree bar chart

This is the code i am trying 
package am;

import java.awt.Color;
import java.awt.Font;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.Properties;

import org.jfree.chart.ChartFactory;
import org.jfree.chart.ChartUtilities;
import org.jfree.chart.JFreeChart;
import org.jfree.chart.axis.AxisLocation;
import org.jfree.chart.axis.NumberAxis;
import org.jfree.chart.labels.ItemLabelAnchor;
import org.jfree.chart.labels.ItemLabelPosition;
import org.jfree.chart.labels.StandardCategoryItemLabelGenerator;
import org.jfree.chart.plot.CategoryPlot;
import org.jfree.chart.plot.PlotOrientation;
import org.jfree.chart.plot.XYPlot;
import org.jfree.chart.renderer.category.BarRenderer;
import org.jfree.chart.renderer.category.CategoryItemRenderer;
import org.jfree.chart.title.LegendTitle;
import org.jfree.data.jdbc.JDBCCategoryDataset;
import org.jfree.ui.RectangleEdge;
import org.jfree.ui.TextAnchor;

public class Monthly_Accounting_Bar {

	public static void main(String[] args) throws Exception {

		String user, host, pwd;

		Properties prop = new Properties();
		FileInputStream fis;
		try {
			fis = new FileInputStream("properties\\server_details.properties");
			prop.load(fis);
			Class.forName(prop.getProperty("driver"));
			host = prop.getProperty("url");
			user = prop.getProperty("username");
			pwd = prop.getProperty("password");

			Connection connect = DriverManager.getConnection(host, user, pwd);

			System.out.println("Connection Successfull");

			String query = "select empname, empid from employee where empcity in (?,?)............;
			PreparedStatement pstmt = connect.prepareStatement(query);

			pstmt.setString(1, "x");
			pstmt.setString(2, "y");

			ResultSet resultSet = pstmt.executeQuery();

			JDBCCategoryDataset dataset = new JDBCCategoryDataset(connect);

			System.out.println("Preparing the data for the chart");
			while (resultSet.next()) {
				dataset.setValue((Double) (resultSet.getDouble("empname")), (resultSet.getString("empid")), "");

			}
			
			System.out.println("checking dataset null condition");

			**if (!(resultSet.next())||dataset==null||dataset.getRowCount==0) {
				System.out.println("dataset  null");

				return;

			} else {
				System.out.println("chart generation");**
				JFreeChart chart = ChartFactory.createBarChart("CHART", "", "", dataset,
						PlotOrientation.HORIZONTAL, true, true, false);
				System.out.println("chart generation done");
				CategoryPlot plot = chart.getCategoryPlot();

				// To display value on top of the chart
				CategoryItemRenderer renderer = ((CategoryPlot) chart.getPlot()).getRenderer();

				renderer.setBaseItemLabelGenerator(new StandardCategoryItemLabelGenerator());
				renderer.setBaseItemLabelsVisible(true);
				ItemLabelPosition position = new ItemLabelPosition(ItemLabelAnchor.OUTSIDE12, TextAnchor.TOP_CENTER);
				renderer.setBasePositiveItemLabelPosition(position);

				BarRenderer renderer1 = (BarRenderer) plot.getRenderer();

				renderer1.setSeriesPaint(0, new Color(255, 210, 59, 255)); // blue
				renderer1.removeChangeListener(plot);

				renderer1.setDrawBarOutline(false);
				renderer1.setItemMargin(0.01);
				chart.setBorderVisible(false);

				chart.setBackgroundPaint(Color.white);
				chart.setBorderPaint(new Color(227, 229, 226));
				chart.getTitle().setFont(new Font("Calibri", Font.BOLD, 20));
				chart.getTitle().setPaint(new Color(89, 89, 89));

				LegendTitle legend = chart.getLegend();
				legend.setPosition(RectangleEdge.RIGHT);
				legend.setBackgroundPaint(Color.WHITE);
				legend.setBorder(0, 0, 0, 0);

				CategoryPlot categoryPlot = chart.getCategoryPlot();
				BarRenderer br = (BarRenderer) categoryPlot.getRenderer();
				br.setMaximumBarWidth(.35); // set maximum width to 25% of chart

				categoryPlot.setRangeAxisLocation(AxisLocation.BOTTOM_OR_RIGHT);

				System.out.println("saving chart in specified folder");
				try {
					ChartUtilities.saveChartAsJPEG(new File(
							"barchart.jpeg"),
							chart, 1300, 200);

					System.out.println("chart is saved successfully.");

				} catch (IOException e) {
					System.out.println("Problem in creating chart.");

				}

			}
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

**Comment by trashgod** [comment]
Your example constructs a [`JDBCCategoryDataset`](https://www.jfree.org/jfreechart/javadoc/org/jfree/data/jdbc/JDBCCategoryDataset.html), but it never invokes `executeQuery()`, either explicitly or via the relevant constructor. As a result, I would expect your `dataset` to remain empty.

**Reply by nagamalli148** [reply]
Thank you so much for the reply.

Actually for this, i think i just made it using the condition like :
if the count of that particular column is null 

eg: if(resultSet.getString("column")==null)
return ;

**Reply by trashgod** [reply]
I don't see a way to examine the `ResultSet` used internally by `JDBCCategoryDataset`. You can check either `getRowCount()` or `getColumnCount()` as a proxy. Regarding your query, note that "The first column will be the category name and remaining columns values (each column represents a series)." Some examples may be found [here](https://stackoverflow.com/search?tab=votes&q=%5bjfreechart%5d%20JDBCCategoryDataset).

**Reply by nagamalli148** [reply]
Yeah I tried before with get row count and with dataset somehow it didn't worked.

So, Finally i got this by doing simple condition if (resultset.next()== null) or if(!resultSet.next()) , used it before generating chart.
This worked as i expected, not generating the chart if the dataset is empty.

anyways thanks for your efforts :)
